### PR TITLE
Bring latest commits from upstream project and delete the .idea folder

### PR DESCRIPTION
**Pull request by felladrin**
## Description

Two issues were causing the project to fail to start up with Maven Wrapper correctly detected:
- We were using an outdated version of Maven Wrapper
- Committing anything in the .idea folder will cause IntelliJ to not auto-configure Maven.
    - We tried leaving the `.idea/runConfigurations` folder in the repo, for example.

I've tried doing one or another, but the only way the project worked correctly with Maven Wrapper was when I applied both changes.

As we noticed the project was working fine with the [upstream Spring-PetClinic](https://github.com/spring-projects/spring-petclinic), we decided to bring the updates that were missing in this repository.

The run configuration was left under `.run` folder due to this [Blog Post from JetBrains](https://blog.jetbrains.com/idea/2020/03/intellij-idea-2020-1-beta2/):
> Note that if you’ve added the `.idea` folder to `.gitignore`, the IDE will suggest you store your run configurations in the `…/project_folder/.run/` directory by default.

ℹ️  This PR should not be squashed upon merge; otherwise, we'll lose the history, making it difficult to sync with the [upstream repository](https://github.com/spring-projects/spring-petclinic) later.

## Repository root files comparison

| [Before](https://github.com/gitpod-io/spring-petclinic/tree/f55e8a2dfcd0f97c3405f7a20fc1a1cfeb8f16fd) | [After](https://github.com/gitpod-io/spring-petclinic/tree/vn/update-from-upstream) |
| --- | --- |
| image |  image |

## Note about a known issue

In IntelliJ, sometimes you may see this warning, but it's harmless and has already been scheduled for fix ([IDEA-300044](https://youtrack.jetbrains.com/issue/IDEA-300044/Opening-new-project-gets-warning-Cannot-resolve-Maven-home-Use-Maven-wrapper-is-not-correct-maven-home-reverting-to-embedded)), so you can ignore it:

image

## How to test

1. Open a workspace, on <gitpod.io>, with IntelliJ IDEA selected as preferred IDE, from the branch of this PR: https://gitpod.io/#https://github.com/gitpod-io/spring-petclinic/tree/vn/update-from-upstream
    a. Or open it from JetBrains Gateway:
    image
2. As it was [prebuilt](https://github.com/gitpod-io/spring-petclinic/pull/67#issuecomment-1254903974), confirm if the project starts fast (i.e. it doesn't need to download the dependencies before starting the PetClinic server), and if "petclinic" is correctly identified as a Maven project, on the sidebar:
    image
3. Open these two workspaces in VS code, to confirm they are rendering the same website:
    - https://gitpod.io/#https://github.com/spring-projects/spring-petclinic
    - https://gitpod.io/#https://github.com/gitpod-io/spring-petclinic/tree/vn/update-from-upstream
    image
4. Confirm if clicking the Debug Button, on 'Run Configuration' menu, allows debugging the app properly.
    image

### CODE REVIEW

**Comment by felladrin** [comment]
I've confirmed the dependencies are being downloaded during the prebuild:
image

**Review by andreafalzetti — APPROVED** [review]
Tested and works as expected, the website loads correctly, the maven project was detected correctly too.

**Review by andreafalzetti — APPROVED** [review]
✅ maven wrapper successfully installed
✅ maven cmds work (e.g. validate, install)
✅ run configuration works (debugger)
✅ app runs

**Comment by felladrin** [comment]
After some more investigation, we've noticed the following:

In the current state of this PR, the Maven libraries are not indexed during the prebuild (where we run the IDE Warmup).

To be able to make maven libraries be indexed (and make the workspace start faster), we need to select the option "_Store generate project files externally_" (in _Preferences | Build, Execution, Deployment | Build Tools | Maven | Importing_) and commit the files it adds to ".idea" folder.  
But making this brings another problem: it breaks the Run Configuration functionality because whenever we add anything inside .idea folder, it doesn't use the Maven Wrapper but the Bundled Maven instead. (This is the same issue we have now on the main branch of this repository)

**td;dr**:
- When we have .idea folder in the repository, Run Configurations doesn't work.
- When don't we have .idea folder in the repository, our workspace takes longer to start, due to the need to index the maven modules.

We're currently discussing this issue about Maven not detecting the Wrapper when .idea folder is present in the repository. [[Internal Link](https://gitpod.slack.com/archives/C02BRJLGPGF/p1663933461700589)]

File: src/main/java/org/springframework/samples/petclinic/vet/Specialty.java [hunk 1]
```diff
@@ -29,6 +27,6 @@
  */
 @Entity
 @Table(name = "specialties")
-public class Specialty extends NamedEntity implements Serializable {
+public class Specialty extends NamedEntity {
```
Comment (svenefftinge) [inline src/main/java/org/springframework/samples/petclinic/vet/Specialty.java:30]: Great!

**Comment by felladrin** [comment]
I'm closing this for now and will get back to it after investigating how to make Warmup+RunConfigurations work.

**Pull request closed without merging**